docs/src/references/c8y-configuration-management.md

```diff
@@ -130,15 +130,25 @@ The `c8y_configuration_plugin` configuration is stored by default under `/etc/te
 
 This [TOML](https://toml.io/en/) file defines the list of files to be managed from the cloud tenant.
 Each configuration file is defined by a record with:
-* The full `path` to the file.
-* An optional configuration `type`. If not provided, the `path` is used as `type`.
-  This `type` is used to declare the configuration file to Cumulocity and then to trigger operations on that file.
-  All the configuration `type`s for the main device are declared to the cloud on start
-  and on change of the `c8y/c8y-configuration-plugin.toml` file.
-* Optional unix file ownership: `user`, `group` and octal `mode`.  
+* The full `path` to a configuration file, OR
+* a `type` which is a unique id for a configuration type supported by the device when the configuration is not a file.
+* Either a `type` or a `path` is mandatory for each record.
+* The `type` or `path` is mapped to a `c8y-config-type` to declare the supported configuration file list to the cloud.
+  The reported `c8y-config-type` is what's visible in the cloud, to trigger operations on each configuration file.
+  All the `c8y-config-type`s for the main device are declared to the cloud on start
+  and on changes of the `c8y/c8y-configuration-plugin.toml` file.
+* If a `type` is specified, it's mapped directly to `c8y-config-type`, else the `path` is used.
+* When a `path` is specified, additional optional unix file ownership paramaters: `user`, `group` and octal `mode` are supported.
   These are only used when a configuration file pushed from the cloud doesn't exist on the device.
   When a configuration file is already present on the device, this plugin never changes file ownership,
   ignoring these parameters.
+* If only the `type` is specified without a `path`, the plugin can not read/update a file on its own.
```

Comment:
> With that we lose here the freedom to add configuration files to c8y-config plugin from cloud site, because you force the chid-device agent (piece of software running on the external child-device) to have all knowledge about paths.

Expecting the child device to know the files that can be remotely managed on it is the right thing to do, isn't it? I feel it's not even a secure approach to let any file be fetched/updated from the child device via the cloud by just updating that toml file, without consulting with the child device if it can handle remote config management on that file. So, even when we update that toml file from the cloud, I'd pass that updated file to the child device and let it confirm if the updates are acceptable to it. For that I expect the child device to know the files that can be remotely managed rather than expecting "any file" to be managed from the cloud. I wouldn't want someone from the cloud randomly fetching some secure key files from the device, without its permission.

> However there is still the question what could be the indication to use filetransfer. Maybe a protocol prefix (`http://`) in the path field, or a field `protocol="http"` can do the job.

I had considered that option too, to have protocol prefixes in the path value like `registry://`, `http://`, `docker://` and `file://` which could be the default. But then I had an alternate thought: why should we enforce such a scheme to the child agent? The child agent is the one declaring these paths/types. So, when it declares these paths, it would know how to read/update them as well, right? Why should it inform thin-edge on what protocol it would use to manage these  paths?

One the question of whether to use HTTP or the file system to upload/share the files, here also I'd leave that to the child agent rather than enforcing it to do it one way or the other. If the child-agent is a local one, it is free to upload the file either using HTTP or via the file system. Let that agent developer choose how he'd wanna do it.